### PR TITLE
New/search

### PR DESCRIPTION
**Pull request by yleaxeman**

### CODE REVIEW

File: src/MOCK_DB_DATA/library-items/search/search.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  return ``;
+}
+
+/*
+ * Return empty string to use a copy of exampleHtml as codeHtml.
+ * Return null to remove Code from library-item.
+ */
+function getCodeHtml(): string | null {
+  return `
+<div class="fhi-search"><!-- add fhi-search--xl for tall search field -->
```
Comment (proand) [inline src/MOCK_DB_DATA/library-items/search/search.ts:27]: This should be the angular example...

File: src/app/views/developer/developer.component.ts [hunk 2]
```diff
@@ -73,7 +73,7 @@ export class DeveloperComponent implements OnInit, OnDestroy {
     }
     switch (this.currentTopLevelMenuItem.name) {
       case TopLevelMenuItemName.components:
-        return this.getComponensMenu();
+        return this.getComponentsMenu();
```
Comment (proand) [inline src/app/views/developer/developer.component.ts:76]: Thanks 👍

File: src/app/views/shared/dynamic-library-examples/fhi-angular-examples/search-example/search-example.component.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
...'Papua New Guinea', 'Paraguay', 'Peru', 'Philippines', 'Pitcairn', 'Poland', 'Portugal', 'Puerto Rico', 'Qatar', 'Reunion', 'Romania', 'Russian Federation', 'Rwanda', 'Saint Kitts and Nevis', 'Saint Lucia', 'Saint Vincent and The Grenadines', 'Samoa', 'San Marino', 'Sao Tome and Principe', 'Saudi Arabia', 'Senegal', 'Seychelles', 'Sierra Leone', 'Singapore', 'Slovak Republic', 'Slovenia', 'Solomon Islands', 'Somalia', 'South Africa', 'S. Georgia and S. Sandwich Isls.', 'Spain', 'Sri Lanka', 'St. Helena', 'St. Pierre and Miquelon', 'Sudan', 'Suriname', 'Svalbard and Jan Mayen Islands', 'Swaziland', 'Sweden', 'Switzerland', 'Syria', 'Taiwan', 'Tajikistan', 'Tanzania', 'Thailand', 'Togo', 'Tokelau', 'Tonga', 'Trinidad and Tobago', 'Tunisia', 'Turkey', 'Turkmenistan', 'Turks and Caicos Islands', 'Tuvalu', 'Uganda', 'Ukraine', 'United Arab Emirates', 'United Kingdom (Britain / UK)', 'United States of America (USA)', 'US Minor Outlying Islands', 'Uruguay', 'Uzbekistan', 'Vanuatu', 'Vatican City State (Holy See)', 'Venezuela', 'Viet Nam', 'Virgin Islands (British)', 'Virgin Islands (US)', 'Wallis and Futuna Islands', 'Western Sahara', 'Yemen', 'Yugoslavia', 'Zaire', 'Zambia', 'Zimbabwe'];
```
Comment (proand) [inline src/app/views/shared/dynamic-library-examples/fhi-angular-examples/search-example/search-example.component.ts:6]: ...but no need to break line here, ok when just dummy data ;)

File: src/app/views/shared/dynamic-library-examples/fhi-angular-examples/search-example/search-example.component.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<div class="fhi-search mb-5">
+  <label for="typeahead-basic" class="form-label">Search for a country (or territory)</label>
+
+  <div class="d-flex">
+    <div class="w-100 position-relative">
+      <input id="typeahead-basic" type="search" class="form-control fhi-search__form-control" placeholder="Start searching..." [(ngModel)]="model" [ngbTypeahead]="search"/>
```
[the remark this answers is not in the document]
Comment (yleaxeman) [reply src/app/views/shared/dynamic-library-examples/fhi-angular-examples/search-example/search-example.component.html:6]: What's the rule?

[the remark this answers is not in the document]
Comment (proand) [reply src/app/views/shared/dynamic-library-examples/fhi-angular-examples/search-example/search-example.component.html:6]: We havent' established a hard rule... we probably should. Many projects use 140, I think 100 is enough...

[the remark this answers is not in the document]
Comment (proand) [reply src/app/views/shared/dynamic-library-examples/fhi-angular-examples/search-example/search-example.component.html:6]: ...and aligning the attributes when more than one line.

File: src/MOCK_DB_DATA/library-items/search/search.ts (hunk 1 above)
Comment (yleaxeman) [reply src/MOCK_DB_DATA/library-items/search/search.ts:27]: Like this?...

File: src/app/views/shared/dynamic-library-examples/fhi-angular-examples/search-example/search-example.component.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (yleaxeman) [reply src/app/views/shared/dynamic-library-examples/fhi-angular-examples/search-example/search-example.component.html]: Is there some kind of config that could take care of this?

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "fhi-frontend-library-web-client",
-  "version": "0.11.1",
+  "version": "3.8.0",
```
Comment (proand) [inline package.json:3]: Skip this... I think the version should be 1.0.0 when we do the "relaunch" of the FHI Designsystem...

**Review by proand — CHANGES_REQUESTED** [review]

**Review by proand — APPROVED** [review]
👍

**Pull request merged**